src/Logic.hs

```diff
@@ -205,7 +205,7 @@ ensureCloned config =
 
 data Event
   -- GitHub events
-  = PullRequestOpened PullRequestId Branch Sha Text Username -- PR, branch, sha, title, author.
+  = PullRequestOpened PullRequestId Branch Branch Sha Text Username -- PR, branch, sha, title, author.
```

Comment:
I see that several constructors and functions are getting two `Branch` arguments. One for the to-be-merged branch, and one for the base branch. To prevent mixing the two arguments, would it be an idea to create a `newtype BaseBranch`? 

On one hand a base branch is conceptually also `Branch`, but on the other hand there's also a `RemoteBranch` already 🤔

This is an open question, having two `Branch` parameters might not be _that_ bad. Having a `BaseBranch` would probably mean some conversion function would need to exist to convert it to `Branch`. It's a tradeoff.

Reply:
`RemoteBranch` is for the thingies that absolutely have to reference something in `origin`. I have not yet reviewed (I thought it was still a draft), so I am not sure if this is one of such cases, but it could be

Reply:
I think `RemoteBranch` is not needed for this case.

Making a new type for the base branch would make things clearer, but I'm not sure whether it's worth to complicate things.

Reply:
> Making a new type for the base branch would make things clearer, but I'm not sure whether it's worth to complicate things.

My 2 cents would be that it would be worth it, either a type-alias or a newtype. I prefer the newtype, because not only does the type document itself, but it prevents you from accidentally swapping the arguments. You can now swap the arguments and the type-system will not catch it. If we add `newtype`, than it would be a type-error to get the arguments mixed-up.

Reply:
Good point!